nstar/src/main.rs

```diff
@@ -12,141 +12,250 @@
 //   See the License for the specific language governing permissions and
 //   limitations under the License.
 
-use anyhow::{Context, Result};
-use colored::Colorize;
-use commands::{Northstar, NstarOpt, Prompt, PromptCommand};
-use futures::StreamExt;
+use anyhow::{anyhow, Context, Error, Result};
+use api::{client::Client, model::Message};
+use futures::{sink::SinkExt, StreamExt};
 use northstar::api::{
     self,
-    client::Client,
-    model::{Container, Request, Response},
+    model::{Container, Request, Version},
+};
+use std::{convert::TryFrom, path::PathBuf, process, str::FromStr, time};
+use structopt::{clap, clap::AppSettings, StructOpt};
+use tokio::{
+    fs,
+    io::{copy, AsyncBufReadExt, BufReader},
 };
-use rustyline::error::ReadlineError;
-use std::{io::Write, path::Path};
-use structopt::StructOpt;
-use tokio::{select, time};
 
-mod commands;
 mod pretty;
-mod terminal;
 
-/// Same stuff but returns a json string
-async fn send_request(
-    client: &mut Client,
-    command: Northstar,
-) -> Result<Response, api::client::Error> {
-    let request = match command {
-        Northstar::Containers => Request::Containers,
-        Northstar::Repositories => Request::Repositories,
-        Northstar::Mount { name, version } => Request::Mount(vec![Container::new(name, version)]),
-        Northstar::Umount { name, version } => Request::Umount(Container::new(name, version)),
-        Northstar::Start { name, version } => Request::Start(Container::new(name, version)),
-        Northstar::Stop {
-            name,
-            version,
-            timeout,
-        } => Request::Stop(Container::new(name, version), timeout),
-        Northstar::Install { npk, repo_id } => {
-            let npk = Path::new(&npk).canonicalize()?;
-            return match client.install(&npk, &repo_id).await {
-                Ok(()) => Ok(Response::Ok(())),
-                Err(e) => Ok(Response::Err(api::model::Error::Npk(e.to_string()))),
-            };
-        }
-        Northstar::Uninstall { name, version } => Request::Uninstall(Container::new(name, version)),
-        Northstar::Shutdown => Request::Shutdown,
-    };
+// About help string generated by build.rs because of static requirements
+const ABOUT: &str = env!("NSTAR_ABOUT");
```

Comment:
Just curious, why is this about taken from an environment variable defined in the `build.rs`?

Reply:
The reason is that the `structopt` about needs a string with static lifetime. The version is obtained via `northstar::api::model::version` which is not `'static`. The env variable is one way to solve this. Otherwise we could use `lazy_static` but I wanted to avoid the additional crate dep.

Reply:
Got it! There is this trick to get a `'static` life time out of things, I just tried it and it works, maybe you can replace this with:

```
fn about() -> &'static str {
    Box::leak(Box::new(format!("Nstar version {}", version())))
}

...
#[structopt(name = "nstar", author, about = about(), global_setting(AppSettings::ColoredHelp))]
```

and remove `build.rs`. It would be a bit simpler I think

Reply:
Ah. Thanks. That makes sense! Will add this.